examples/app/export_csv/download.js

```diff
@@ -1,14 +1,23 @@
-var data = source.data;
-var filetext = 'name,income,years_experience\n';
-for (var i = 0; i < data['name'].length; i++) {
-    var currRow = [data['name'][i].toString(),
-                   data['salary'][i].toString(),
-                   data['years_experience'][i].toString().concat('\n')];
+function table_to_csv(data_table) {
+    var columns = Object.keys(data_table)
+    var nrows = data_table[columns[0]].length
```

Comment:
CDS has a `get_length` method, I think it would be better to use that, though it would require `table_to_csv` to accept `source` and not `source.data`

Reply:
I must have missed that, I don't see anything about a ColumnDataSource get_length method [in the docs](https://bokeh.pydata.org/en/latest/docs/reference/models/sources.html)?

It's implemented now.